src/wwwroot/template-editor/adapterService.js

```diff
@@ -0,0 +1,356 @@
+(function() {
+    'use strict';
+  
+    angular
+      .module('mseditor')
+      .service('adapterService', adapterService);
+  
+      adapterService.$inject = [
+      '$http',
+      '$q'
+    ];
+  
+    function adapterService($http, $q) {
+      var service = {
+        getCampaign: getCampaign,
+        campaignSaveChanges: campaignSaveChanges,
+        campaignSaveChangesAsTemplate: campaignSaveChangesAsTemplate,
+        getCampaignRssPreview: getCampaignRssPreview,
+        deleteImages: deleteImages,
+        getImagesForCampaign: getImagesForCampaign,
+        getPreviewImage: getPreviewImage,
+        getPreviewImageWithPlay: getPreviewImageWithPlay,
+        getTiendaNubeProducts: getTiendaNubeProducts,
+        getSettings: getSettings
+      };
+  
+      return service;
+
+      /**
+       * Return a preview placeholder image
+       * @method
+       * @argument {object} params
+       * @example
+       * {
+       *  fileName: fileName,
+       *  providerName: providerName,
+       *  idCampaign: campaign.getIdCampaign(),
+       *  idTemplate: campaign.getIdTemplate()
+       * }
+       */
+      function getPreviewImageWithPlay(params) {
+        console.log("getPreviewImageWithPlay");
+        var response = {
+          ThumbnailUrlWithPlay: 'https://i.vimeocdn.com/filter/overlay?src0=' + 'http://www.wpclipart.com/travel/US_Road_Signs/info/temporary.png' + '&src1=http%3A%2F%2Ff.vimeocdn.com%2Fp%2Fimages%2Fcrawler_play.png'
+        };
+        var fakePromise = $q.resolve(response);
+        return fakePromise;
```

Comment:
why not simply `return $q.resolve(response);`? (I am just asking, I am not suggesting a change)

Reply:
I think it is more declarative.

Reply:
What does that mean? and why is that a good thing?

Reply:
I think that that people need to learn the basis of promises and then come back.

Promises are today a well-known standard, and `$q` is a friendly name as `$` was for `jQuery` or `_` is for Underscore or LoadDash libraries.

In our context `return $q.resolve(response);` is very similar to `return Task.FromResult(response);` in a C# project. I really do not see any advantage on:

```cs
var fakeTask = Task.FromResult(response);
return fakeTask;
```

over 

```cs
return Task.FromResult(response);
```

Reply:
If you want to be clear that the response is faked, maybe it is better idea using _fake_ prefix on _response_, for example:

```js
var fakeResponse = {
    // . . . 
};
return $q.resolve(fakeResponse);
```

Since our implementation is completely based on responses, they should be as less intrusive and less verbose as possible. If we would using [Kris Kowal's promises](https://github.com/kriskowal/q) it would be shortner yet: `return q(fakeResponse);`

Reply:
If you want to be clear that the response is faked, maybe it is better idea using _fake_ prefix on _response_, for example:

```js
var fakeResponse = {
    // . . . 
};
return $q.resolve(fakeResponse);
```

Since our implementation is completely based on **promises**, they should be as less intrusive and less verbose as possible. If we would using [Kris Kowal's promises](https://github.com/kriskowal/q) it would be shortner yet: `return q(fakeResponse);`

Reply:
Ok, I see your point. 👍  thanks andres :)